pkg/azuredisk/device_perf_optimization_linux.go

```diff
@@ -0,0 +1,296 @@
+// +build linux
+
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuredisk
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+type DeviceHelper struct{}
+
+func (deviceHelper *DeviceHelper) OptimizeDevicePerformance(nodeInfo *NodeInfo, diskSkus map[string]map[string]DiskSkuInfo, devicePath string, attributes map[string]string) (err error) {
+
+	klog.V(2).Infof("OptimizeDevicePerformance: Tuning settings for %s", devicePath)
+
+	if nodeInfo == nil {
+		err = fmt.Errorf("Node info is not provided. Error: invalid parameter")
+		klog.Errorf("Error: %s", err)
```

Comment:
no need to log error everywhere, just `return err` and could log error in outside function, then only need to write one logging which would make code clean

Reply:
Changing this behavior for places where I am returning custom errors. If I don't log the errors returned by routines then it would be hard to debug the issues. given we will not have the line no. etc in the calling function.

Reply:
`return fmt.Errorf("Node info should not be nil")` is ok, it already tells complete info, don't need log where error happens in the code